File: strings/src/test/java/org/example/strings/PermutationTest.java
package org.example.strings;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.example.strings.Permutation.isPermutation;

public class PermutationTest {

    @Test
    public void equalStringsArePermutations() {
        assertThat(isPermutation("dog", "dog")).isTrue();
    }

    @Test
    public void permutationAllowsSpaces() {
        assertThat(isPermutation("my dog", "my dog")).isTrue();
    }

    @Test
    @Disabled
    public void permutationIsCaseSensitive() {
        assertThat(isPermutation("dog", "Dog")).isFalse();
    }

    @Test
    @Disabled
    public void permutationsAreEqualInLength() {
        assertThat(isPermutation("game", "games")).isFalse();
        assertThat(isPermutation("games", "game")).isFalse();
    }

    @Test
    @Disabled
    public void sameSizedStringsAreNotPermutations() {
        assertThat(isPermutation("mace", "mage")).isFalse();
        assertThat(isPermutation("mage", "mace")).isFalse();
    }

    @Test
    @Disabled
    public void permutationOfUnequalStrings() {
        assertThat(isPermutation("game", "mage")).isTrue();
        assertThat(isPermutation("mage", "game")).isTrue();
    }

    @Test
    @Disabled
    public void duplicateLettersThatAreNotPermutations() {
        assertThat(isPermutation("magge", "gaame")).isFalse();
        assertThat(isPermutation("gaame", "magge")).isFalse();
    }

    @Test
    @Disabled
    public void spacesWithUnequalNumberOfSpaces() {
        assertThat(isPermutation("ga  me", "mage ")).isFalse();
        assertThat(isPermutation("mage ", "ga  me")).isFalse();
    }

}

File: strings/csharp/StringsTest/PermutationTest.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;
using static StringHomework.Permutation;

namespace StringHomeworkTest
{
    internal class PermutationTest
    {
        [Test]
        public void EqualStringsArePermutations()
        {
            Assert.IsTrue(IsPermutation("dog", "dog"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void PermutationIsCaseSensitive()
        {
            Assert.IsFalse(IsPermutation("dog", "Dog"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void PermutationsAllowSpaces()
        {
            Assert.IsTrue(IsPermutation("my dog", "my dog"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void PermutationsEqualInLength()
        {
            Assert.IsFalse(IsPermutation("game", "games"));
            Assert.IsFalse(IsPermutation("games", "game"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void SameSizedStringsAreNotPermutations()
        {
            Assert.IsFalse(IsPermutation("mace", "mage"));
            Assert.IsFalse(IsPermutation("mage", "mace"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void PermutationOfUnequalStrings()
        {
            Assert.IsTrue(IsPermutation("game", "mage"));
            Assert.IsTrue(IsPermutation("mage", "game"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void DuplicateLettersThatAreNotPermutations()
        {
            Assert.IsFalse(IsPermutation("gaame", "magge"));
            Assert.IsFalse(IsPermutation("magge", "gaame"));
        }

        [Test]
        [Ignore("Implement this feature")]
        public void SpacesWithUnequalNumberOfSpaces()
        {
            Assert.IsFalse(IsPermutation("mage ", "ga  me"));
            Assert.IsFalse(IsPermutation("ga  me", "mage "));
        }
    }
}
